### PR TITLE
website: Document Vagrant Cloud search API

### PR DESCRIPTION
**Pull request by justincampbell**

### CODE REVIEW

File: website/source/docs/vagrant-cloud/api.html.md.erb [hunk 1]
```diff
@@ -530,6 +530,101 @@ Sends a 2FA code to the requested delivery method.
...
+
+`GET /api/v1/search`
+
+#### Arguments
+
+* `q` - (Optional) The search query. Results will match the `username`, `name`, or `short_description` fields for a box. If omitted, the top boxes based on `sort` and `order` will be returned (defaults to "downloads desc").
+* `provider` - (Optional) Filter results to boxes supporting for a specific provider.
+* `sort` - (Optional, default: `"downloads"`) The field to sort results on. Can be one of `"downloads"`, `"created"`, or `"updated"`.
+* `order` - (Optional, default: `"desc"`) The order to return the sorted field in. Can be `"desc"` os `"asc"`.
+* `limit` - (Optional, default: `10`) The number of results to return (max of TODO).
```
Comment (briancain) [inline website/source/docs/vagrant-cloud/api.html.md.erb]: `max of TODO` should be fixed to whatever the max is

File: website/source/docs/vagrant-cloud/api.html.md.erb [hunk 2]
```diff
@@ -530,6 +530,101 @@ Sends a 2FA code to the requested delivery method.
...
+
+### Search for boxes
+
+`GET /api/v1/search`
+
+#### Arguments
+
+* `q` - (Optional) The search query. Results will match the `username`, `name`, or `short_description` fields for a box. If omitted, the top boxes based on `sort` and `order` will be returned (defaults to "downloads desc").
+* `provider` - (Optional) Filter results to boxes supporting for a specific provider.
+* `sort` - (Optional, default: `"downloads"`) The field to sort results on. Can be one of `"downloads"`, `"created"`, or `"updated"`.
```
Comment (briancain) [inline website/source/docs/vagrant-cloud/api.html.md.erb]: Maybe this is a better question for the corresponding website PR, but what gets returned when you provide a sort key that isn't supported? Is there a new HTTP response code that needs to be documented here?

File: website/source/docs/vagrant-cloud/api.html.md.erb [hunk 3]
```diff
@@ -530,6 +530,101 @@ Sends a 2FA code to the requested delivery method.
...
+    end
+    ```
+  </div>
+</div>
+
+#### Example Response
+
+```json
+{
+  "boxes": [
```
Comment (briancain) [inline website/source/docs/vagrant-cloud/api.html.md.erb]: Does this response body also need to include `downloads`?

File: website/source/docs/vagrant-cloud/api.html.md.erb (hunk 2 above)
Comment (justincampbell) [reply website/source/docs/vagrant-cloud/api.html.md.erb]: I think right now it just ignores it, but I will check.

Comment (justincampbell) [reply website/source/docs/vagrant-cloud/api.html.md.erb]: Validated now! Will return 400.

**Comment by briancain** [comment]
Looking good, I think we're just missing the new HTTP 400 status code documented with the rest of the response codes.

**Pull request merged**